vllm/v1/request.py

```diff
@@ -28,17 +29,22 @@ def __init__(
         sampling_params: SamplingParams,
         eos_token_id: Optional[int],
         client_index: int = 0,
+        arrival_time: Optional[float] = None,
         lora_request: Optional["LoRARequest"] = None,
         structured_output_request: Optional["StructuredOutputRequest"] = None,
         cache_salt: Optional[str] = None,
+        priority: int = 0,
     ) -> None:
         self.request_id = request_id
         self.client_index = client_index
+        self.priority = priority
         self.sampling_params = sampling_params
         # Because of LoRA, the eos token id can be different for each request.
         self.eos_token_id = eos_token_id
         self.lora_request = lora_request
         self.structured_output_request = structured_output_request
+        self.arrival_time = arrival_time if arrival_time is not None else \
+            time.time()
```

Comment:
I have a small concern on this: There are some "niche" cases (mostly for veRL integration IIRC) where vLLM uses "SPMD scheduler." In this case, a scheduler is co-located with each worker and we rely on the assumption that every scheduler will make the same scheduling decision every step. In this case, using the system time (`time.time()`) for scheduling will lead to a critical bug.

cc @youkaichao In the SPMD scheduler case, does each scheduler receive the same `EngineCoreOuptut` (so that we can rely on its `arrival_time`)?
Otherwise, I think we should detect this case and raise an error when initializing the vLLM engine.